ota-plus-web/app/com/advancedtelematic/auth/oidc/NamespaceProvider.scala

```diff
@@ -1,16 +1,43 @@
 package com.advancedtelematic.auth.oidc
 
-import javax.inject.Inject
+import akka.http.scaladsl.util.FastFuture
+import com.advancedtelematic.api.{ApiClientExec, ApiClientSupport}
 import com.advancedtelematic.auth.Tokens
 import com.advancedtelematic.libats.data.DataType.Namespace
+import javax.inject.Inject
 import play.api.Configuration
+import play.api.libs.ws.WSClient
 
-trait NamespaceProvider extends (Tokens => Namespace)
+import scala.concurrent.{ExecutionContext, Future}
+
+trait NamespaceProvider extends (Tokens => Future[Namespace])
 
 class NamespaceFromIdentity extends NamespaceProvider {
-  override def apply(tokens: Tokens): Namespace = Namespace(tokens.idToken.userId.id)
+  override def apply(tokens: Tokens): Future[Namespace] = FastFuture.successful{
+    Namespace(tokens.idToken.userId.id)
+  }
 }
 
+class NamespaceFromUserProfile @Inject()(val conf: Configuration,
+                                         val ws: WSClient,
+                                         val clientExec: ApiClientExec)
+                                        (implicit ec: ExecutionContext)
+  extends NamespaceProvider with ApiClientSupport {
+
+  override def apply(tokens: Tokens): Future[Namespace] =
+    userProfileApi.userNamespaces(tokens.idToken.userId).map {
+      case namespaces if namespaces.isEmpty =>
+        Namespace.generate
```

Comment:
What is this doing? Might be needing user-profile to create profile on `UserLogin` event. Probably need to redo that in user-profile.